public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -360,22 +358,32 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
                     }
             }
 
-
             this.conceptSynonymNames.put(con.getConceptId(),conceptSynonymNames);
 
             if((con.getStandardConcept() == null || !con.getStandardConcept().equals("S") ) && (searchConceptsRequest.getQuery().equals(conceptCode) || searchConceptsRequest.getQuery().equals(conceptId))){
                 response.setMatchType(conceptCode.equals(searchConceptsRequest.getQuery()) ? MatchType.CODE : MatchType.ID );
 
                 List<Concept> std_concepts = conceptDao.findStandardConcepts(con.getConceptId());
-                response.setStandardConcepts(std_concepts.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+                List<Concept> stdConceptswithSynonyms = new ArrayList<>();
+                for(Concept concept: std_concepts){
+                    Concept conceptWithSynonyms = conceptDao.fetchConceptWithSynonyms(concept.getConceptId());
+                    stdConceptswithSynonyms.add(conceptWithSynonyms);
+                    ArrayList<String> uniqueConceptSynonymNames = new ArrayList<>();
+                    for(ConceptSynonym cs:conceptWithSynonyms.getSynonyms()){
+                        if(!uniqueConceptSynonymNames.contains(cs.getConceptSynonymName()) && !concept.getConceptName().equals(cs.getConceptSynonymName())){
```

Comment:
So you are just removing the concept name from synonm list right? There really aren't any duplicates ... We can ennforce that on the database level. 

So i would  do this here:  https://github.com/all-of-us/workbench/blob/master/common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java#L230

That function i'm sure wants the duplicate removed.